src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,168 @@
+                } else {
+                    return new FindCommand(args);
+                }
+            } else if (input.isBlank()) {
+                return new ErrorCommand("You need to tell me what you want me to do!");
+            } else {
+                return new ErrorCommand("Sorry, I don't understand that!");
+            }
+        } catch (DukeParsingException e) {
+            return new ErrorCommand(e.getMessage());
+        }
+    }
+
+    // helpers for parse function
+
+    private static void ensureNoArgs(String args, String commandName) throws DukeParsingException {
+        if (!args.isEmpty()) {
+            throw new DukeParsingException(String.format("I don't understand that. Did you mean %s?", commandName));
+        }
+    }
+
+    private static int parseTaskNumber(String args, String taskDescription, String example)
+            throws DukeParsingException {
+        try {
+            return Integer.parseInt(args);
+        } catch (NumberFormatException e) {
+            throw new DukeParsingException(
+                    String.format("You need to tell me the number of the task %s. Eg. %s", taskDescription, example));
+        }
+    }
+
+    private static Event parseEvent(String args) throws DukeParsingException {
+        String[] argsSplit;
+        String description;
+        String[] dateStrings;
+
+        try {
+            argsSplit = splitAround(args, "\\s+/at\\s+");
+            description = argsSplit[0];
+            dateStrings = splitAround(argsSplit[1], "\\s*-\\s*");
+        } catch (DukeParsingException e) {
+            throw new DukeParsingException("Couldn't add event! To add an event, talk to me using "
+                    + "the format event <description> /at <start>-<end>!");
+        }
+
+        Date start = parseDate(dateStrings[0]);
+        Date end = parseDate(dateStrings[1]);
+        if (start.after(end)) {
+            throw new DukeParsingException("Start date is after end date!");
+        }
+        return new Event(description, start, end);
+    }
+
+    private static Deadline parseDeadline(String args) throws DukeParsingException {
+        String[] argsSplit;
+        String description;
+        try {
+            argsSplit = splitAround(args, "\\s+/by\\s+");
+            description = argsSplit[0];
+        } catch (DukeParsingException e) {
+            throw new DukeParsingException("Couldn't add deadline! To add a deadline, talk to me using "
+                    + "the format deadline <description> /by <date>!");
+        }
+        Date date = parseDate(argsSplit[1]);
+        return new Deadline(description, date);
+    }
+
+    /**
+     * Splits a string into 2 around the first occurence of a regex pattern. An exception is thrown if the pattern does
+     * not exist, or if either of the tokens are blank.
+     */
+    private static String[] splitAround(String string, String pattern) throws DukeParsingException {
+        int n = 2;
```

Comment:
Maybe a comment should be used to explain what n represents, and why 2 is used (magic number)